aptos-move/framework/aptos-framework/sources/voting.move

```diff
@@ -283,35 +356,59 @@ module aptos_framework::voting {
         );
     }
 
-    /// Resolve the proposal with given id. Can only be done if there are at least as many votes as min required and
-    /// there are more yes votes than no. If either of these conditions is not met, this will revert.
-    ///
-    /// @param voting_forum_address The address of the forum where the proposals are stored.
-    /// @param proposal_id The proposal id.
-    public fun resolve<ProposalType: store>(
+    fun is_proposal_resolvable<ProposalType: store>(
         voting_forum_address: address,
         proposal_id: u64,
-    ): ProposalType acquires VotingForum {
+    ) acquires VotingForum {
         let proposal_state = get_proposal_state<ProposalType>(voting_forum_address, proposal_id);
         assert!(proposal_state == PROPOSAL_STATE_SUCCEEDED, error::invalid_state(EPROPOSAL_CANNOT_BE_RESOLVED));
 
         let voting_forum = borrow_global_mut<VotingForum<ProposalType>>(voting_forum_address);
         let proposal = table::borrow_mut(&mut voting_forum.proposals, proposal_id);
         assert!(!proposal.is_resolved, error::invalid_state(EPROPOSAL_ALREADY_RESOLVED));
 
+        let multi_step_in_execution_key = utf8(IS_MULTI_STEP_PROPOSAL_IN_EXECUTION_KEY);
+        if (simple_map::contains_key(&proposal.metadata, &multi_step_in_execution_key)) {
+            let is_multi_step_proposal_in_execution_value = simple_map::borrow_mut(&mut proposal.metadata, &multi_step_in_execution_key);
+            *is_multi_step_proposal_in_execution_value = to_bytes(&true);
+        };
+
         // We need to make sure that the resolution is happening in
         // a separate transaction from the last vote to guard against any potential flashloan attacks.
         let resolvable_time = to_u64(*simple_map::borrow(&proposal.metadata, &utf8(RESOLVABLE_TIME_METADATA_KEY)));
         assert!(timestamp::now_seconds() > resolvable_time, error::invalid_state(ERESOLUTION_CANNOT_BE_ATOMIC));
 
-        let resolved_early = can_be_resolved_early(proposal);
-        proposal.is_resolved = true;
-        proposal.resolution_time_secs = timestamp::now_seconds();
-
         assert!(
             transaction_context::get_script_hash() == proposal.execution_hash,
             error::invalid_argument(EPROPOSAL_EXECUTION_HASH_NOT_MATCHING),
         );
+    }
+
+    /// Resolve the proposal with given id. Can only be done if there are at least as many votes as min required and
```

Comment:
Can you add some documentation here to explain that this can only be called for single-step proposals because multi-step proposals need the next execution hash?

Reply:
yup just added!